scheduler/src/cook/kubernetes/api.clj

```diff
@@ -605,6 +605,31 @@
            period-sec
            (assoc "COOK_CHECKPOINT_PERIOD_SEC" (str period-sec))))))))
 
+(def default-checkpoint-failure-reasons
+  "Default set of failure reasons that should be counted against checkpointing attempts"
+  #{:mesos-unknown
+    :mesos-command-executor-failed
+    :mesos-container-limitation-memory
+    :mesos-container-launch-failed})
+
+(defn calculate-effective-checkpointing-config
+  "Given the job's checkpointing config, calculate the effective config. Making any adjustments such as defaults,
+  overrides, or other behavior modifications."
+  [{:keys [job/checkpoint job/instance] :as job} task-id]
+  (when checkpoint
+    (let [{:keys [default-checkpoint-config]} (config/kubernetes)
+          checkpoint (merge default-checkpoint-config (util/job-ent->checkpoint job))]
+      (if-let [max-checkpoint-attempts (:max-checkpoint-attempts checkpoint)]
```

Comment:
Can we destructure this in the `let` above?

Reply:
updated